src/chart_types/xy_chart/rendering/utils.ts

```diff
@@ -67,6 +67,14 @@ export function isDatumFilled({ filled, initialY1 }: DataSeriesDatum) {
 export function getClippedRanges(dataset: DataSeriesDatum[], xScale: Scale, xScaleOffset: number): ClippedRanges {
   let firstNonNullX: number | null = null;
   let hasNull = false;
+
+  const completeDatasetIsNull = dataset.every((datum) => isDatumFilled(datum));
+
+  if (completeDatasetIsNull) {
+    const [min, max] = xScale.range;
+    return [[min, max - xScale.step / 2]];
+  }
```

Comment:
Sorry @anishagg17 I edited this to exclude the step. You are too quick 💨 

```suggestion
  if (completeDatasetIsNull) return [xScale.range];
```